logging/logger_registry.go

```diff
@@ -82,6 +82,12 @@ func (lr *loggerRegistry) updateLoggerLevel(name string, level Level) error {
 }
 
 func (lr *loggerRegistry) updateLoggerRegistry(logConfig []LoggerPatternConfig) error {
+	for _, name := range lr.getRegisteredLoggerNames() {
+		err := lr.updateLoggerLevel(name, INFO)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
:warning: disclaimer, this comment only applies [if we stick with "processing all log changes"](https://github.com/viamrobotics/rdk/pull/4232#issuecomment-2252804245).

let's do the INFO reset after processing changes. i think the current approach has a small race condition where an existing logger configured with a non-INFO level is briefly reset to INFO

Reply:
> after processing changes

Sorry which "processing changes" are you referring to? If we put this block at the end of the function, wouldn't everything get set to `INFO`?

Reply:
sorry, i meant just the loggers that are still in the registry but not in the config

Reply:
more specifically, we should keep a count of how many times each registered pattern is configured by a new config pattern, and then reset all registered loggers with 0 configurations back to INFO.

Reply:
Ah cool; thanks for the clarification.